api/ServerInput.cs

```diff
@@ -1,23 +1,80 @@
-﻿/**
+﻿	/**
  * Input for reading device state from the Server API
  */
 using System;
+using System.Net;
+using System.Threading.Tasks;
+using System.Net.Http;
+using System.Text;
+using System.Diagnostics;
+using PortableRest;
+using Newtonsoft.Json;
+
 
 namespace api
 {
 public class ServerInput : IDeviceInput
 {
-	public ServerInput() 
+	string _server_uri;
+
+	public ServerInput(string server_uri) 
 	{
+		_server_uri= server_uri;
 	}
 
 	/**
 	 * Updates device from the server
 	 */
 	public bool read(Device dev)
-	{
-		//TODO: Implement this function
-		return true;
+	{ 
+		bool server_input_status = false; //The variable tracks the successful updataion of state for the app and decision making
+
+
+		//this try block tries to get a deserialized string of the state from the server
+		try
+		{
+		var client = new RestClient();
+		client.BaseUrl = _server_uri;
+		var request = new RestRequest("api/storage/device/{house}/{space}/{device}");//the default request is GET
+		request.AddUrlSegment("house", dev.ID.HouseID.ToString());
+		request.AddUrlSegment("space", dev.ID.HouseID.ToString());
+		request.AddUrlSegment("device", dev.ID.HouseID.ToString());
+		var response = client.ExecuteAsync<string>(request);//tries to get a deserialized string of device state as the response.
+		 
+			if(response.Status ==TaskStatus.Created)//condition check for a successful completion of the GET request
```

Comment:
Is this the right code? Not an OK or anything else? I haven't double checked the TaskStatus values, but this seems counter-intuitive given that this is a GET request.